JavaScriptCodeConventions.md

```diff
@@ -0,0 +1,84 @@
+# JavaScript Code Conventions
+> NOTE: These conventions also apply to super-sets of JavaScript which are used within Ultimaker such as TypeScript.
+
+## Introduction
+These JavaScript style guidelines are intended to be used by both the R&D and marketing team as part of a broader effort to standardize JavaScript style across the company and as an early facilitator for the fact that Ultimaker continues to offer more web-based products such as the Cura Cloud services and Cura Connect.
+
+## AirBnB as Baseline Style
+To reduce the need to build comprehensive style guidelines ourselves, we have decided to adopt the well-known and widely used **AirBnB style**. The full documentation for AirBnB style [can be found on GitHub](https://github.com/airbnb/javascript). For the most part, this style is already very similar to what is used by front-end teams at Ultimaker with a few differences, which are outlined below. It was also chosen because of it was designed for usage with React.js, which is used by the R&D department. It is applicable to Vue.js as well, however, and can be used by the marketing department.
+
+## Enforcing AirBnB Style in Your Editor
+### Set-Up for VS Code
+1. Install packages:
+
+```
+npm i --save-dev eslint eslint-config-airbnb eslint-plugin-import eslint-plugin-jsx-a11y eslint-plugin-react typescript-eslint-parser
+```
+2. Install the [`eslint` plugin for VS Code](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint).
+3. Create an `.eslintrc.json` file in your project’s root directory.
+4. Add the following content to `.eslintrc.json`:
+
+```
+{
+    "extends": "airbnb",
+    "env": {
+        "node": true,
+        "es6": true,
+        "browser": true
+    },
+    "parserOptions": {
+        "ecmaVersion": 6,
+        "sourceType": "module",
+        "ecmaFeatures": {
+            "modules": true,
+            "jsx": true
+        }
+    },
+    "plugins": ["import"],
+    "rules": {
+        "indent": ["error", 4, {
+            "ignoredNodes": ["JSXElement", "JSXElement > *", "JSXAttribute", "JSXIdentifier", "JSXNamespacedName", "JSXMemberExpression", "JSXSpreadAttribute", "JSXExpressionContainer", "JSXOpeningElement", "JSXClosingElement", "JSXText", "JSXEmptyExpression", "JSXSpreadChild"]
+        }],
+        "no-underscore-dangle": 0,
+        "import/no-unresolved": 0
+    },
+    "settings": {
+        "import/extensions": [".js", ".jsx"]
+    },
+    "overrides": [
+        {
+            "files": ["**/*.ts", "**/*.tsx"],
+            "parser": "typescript-eslint-parser",
+            "rules": {
+                "no-undef": 0,
+                "react/jsx-filename-extension": [1, {
```

Comment:
It would be great to put framework specific implementations in additional files. It could be valid to use a different framework or no framework depending on the projects requirements. This way we can state then when you use the 'wc-eend' framework you also take the wc-eend additions.

Reply:
Sure, we will change this